src/tiled/minimaprenderer.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * minimaprenderer.h
+ * Copyright 2017, Yuriy Natarov <natarur@gmail.com>
+ * Based on a minimap by Christoph Schnackenberg and Thorbjørn Lindeijer
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "mapdocument.h"
+
+#include <QImage>
+
+namespace Tiled {
+namespace Internal {
+
+class MiniMapRenderer
+{
+public:
+    enum MiniMapRenderFlag {
+        DrawObjects             = 0x0001,
+        DrawTiles               = 0x0002,
+        DrawImages              = 0x0004,
+        IgnoreInvisibleLayer    = 0x0008,
+        DrawGrid                = 0x0010
+    };
+
+    Q_DECLARE_FLAGS(MiniMapRenderFlags, MiniMapRenderFlag)
+
+    static MiniMapRenderer* instance();
```

Comment:
I doesn't makes sense to have the `MiniMapRenderer` as a singleton, since it doesn't store any state that would be useful to keep around. Instead, please just allocate it on the stack where you need it, passing the `MapDocument*` to its constructor instead of having a `setMapDocument` method:
```c++
MiniMapRenderer renderer(mMapDocument);
renderer.renderToImage(image, flags);
```